samples/extensions/open_cl_interop/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2021, Arm Limited and Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 the "License";
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+ 
+# The OpenCL interoperability only works on Android
+get_filename_component(FOLDER_NAME ${CMAKE_CURRENT_LIST_DIR} NAME)
+get_filename_component(PARENT_DIR ${CMAKE_CURRENT_LIST_DIR} PATH)
+get_filename_component(CATEGORY_NAME ${PARENT_DIR} NAME)
+
+# OpenCL
+add_library(opencl INTERFACE)
+set(OPENCL_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/third_party/OpenCL-Headers/)

```

Comment:
would it make sense to use FIND_OPENCL?  i.e. try something like this:
find_package(OpenCL);
target_link_libraries(target public OpenCL::OpenCL)

Then you wouldn't need to include OpenCL headers in this PR and if the OpenCL headers installed in the system are updated, CMake will use those (i.e. we won't have to worry about grabbing the latst OpenCL headers

Reply:
I tried to use the same approach as in `Vulkan-Samples/third_party/CMakeLists.txt`.  All the dependencies are included (even vulkan itself) so that the developers don't need to install anything extra. 
If OpenCL is not present for some reason (for example, not supported on the host platform, but available on the target device), then the headers will be missing.

Reply:
It's probably best to include them as a submodule within the third_party folder of the framework so they're exposed to more potential samples in addition it'd be easier to keep it up to date.
However, something I'd like to recommend is to show use of system based inclusion such as the find_package.  If not found (i.e. in cmake after find_package OPENCL_NOTFOUND would be what the variable is set as).  Then fall back to the one in the third_party submodule.
The idea behind this approach is to encourage consumers of the sample to learn best practices and not embed a snap shot in time but rather the package which gets auto-updated with the system.